WordPress/src/main/java/org/wordpress/android/ui/reader/actions/ReaderTagActions.java

```diff
@@ -67,59 +74,79 @@ public void onErrorResponse(VolleyError volleyError) {
         return true;
     }
 
-    public static boolean addTag(final ReaderTag tag,
+    public static boolean addTag(@NotNull final ReaderTag tag,
                                  final ReaderActions.ActionListener actionListener) {
-        if (tag == null) {
-            ReaderActions.callActionListener(actionListener, false);
-            return false;
-        }
+        ReaderTagList tags = new ReaderTagList();
+        tags.add(tag);
+        return addTags(tags, actionListener);
+    }
 
-        final String tagNameForApi = ReaderUtils.sanitizeWithDashes(tag.getTagSlug());
-        final String path = "read/tags/" + tagNameForApi + "/mine/new";
-        String endpoint = "/read/tags/" + tagNameForApi + "/posts";
+    public static boolean addTags(@NotNull final List<ReaderTag> tags) {
+        return addTags(tags, null);
+    }
 
-        ReaderTag newTag = new ReaderTag(
-                tag.getTagSlug(),
-                tag.getTagDisplayName(),
-                tag.getTagTitle(),
-                endpoint,
-                ReaderTagType.FOLLOWED);
+    public static boolean addTags(@NotNull final List<ReaderTag> tags,
+                                 final ReaderActions.ActionListener actionListener) {
+        ReaderTagList existingFollowedTags = ReaderTagTable.getFollowedTags();
+
+        ReaderTagList newTags = new ReaderTagList();
+        for (ReaderTag tag : tags) {
+            final String tagNameForApi = ReaderUtils.sanitizeWithDashes(tag.getTagSlug());
+            String endpoint = "/read/tags/" + tagNameForApi + "/posts";
+
+            ReaderTag newTag = new ReaderTag(
+                    tag.getTagSlug(),
+                    tag.getTagDisplayName(),
+                    tag.getTagTitle(),
+                    endpoint,
+                    ReaderTagType.FOLLOWED);
+            newTags.add(newTag);
+        }
 
-        com.wordpress.rest.RestRequest.Listener listener = new RestRequest.Listener() {
-            @Override
-            public void onResponse(JSONObject jsonObject) {
-                AppLog.i(T.READER, "add tag succeeded");
-                // the response will contain the list of the user's followed tags
-                ReaderTagList tags = parseFollowedTags(jsonObject);
-                ReaderTagTable.replaceFollowedTags(tags);
+        com.wordpress.rest.RestRequest.Listener listener = jsonObject -> {
+            AppLog.i(T.READER, "add tag succeeded");
+            // the response will contain the list of the user's followed tags
+            ReaderTagList followedTags = parseFollowedTags(jsonObject);
+            ReaderTagTable.replaceFollowedTags(followedTags);
+            if (actionListener != null) {
                 ReaderActions.callActionListener(actionListener, true);
             }
```

Comment:
I'd want to remove this action listener used to communicate result when following a single tag (on Topic Details and ReaderSubsActivity). I can do it when I refactor "Topic Details" to support logged out scenario. Or let me know if I should do it in this PR.